classes/install/Upgrade.inc.php

```diff
@@ -1566,35 +1566,51 @@ function convertEditorDecisionNotes() {
 	 */
 	function convertCommentsToEditor() {
 		$submissionDao = Application::getSubmissionDAO();
+		$stageAssignmetDao = DAORegistry::getDAO('StageAssignmentDAO');
 		$queryDao = DAORegistry::getDAO('QueryDAO');
 		$noteDao = DAORegistry::getDAO('NoteDAO');
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+
+		import('lib.pkp.classes.security.Role'); // ROLE_ID_...
 
 		$commentsResult = $submissionDao->retrieve(
-			'SELECT s.submission_id, s.comments_to_ed, s.date_submitted, sa.user_id
+			'SELECT s.submission_id, s.comments_to_ed, s.date_submitted
```

Comment:
I think if you moved the `context_id` into this query as well, then you wouldn't need to instantiate the `Submission` object for these entries. That'll keep the number of DB queries in DAO functions required for this upgrade step to a minimum, which will mean we're less likely to hit an upgrade tangle later if we change schemas (since this upgrade step runs before the schema sync).